orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/client/SaltConnector.java

```diff
@@ -126,15 +128,34 @@ public SaltBootResponses action(SaltAction saltAction) {
     }
 
 
-    public <T> T run(Target<String> target, String fun, SaltClientType clientType, String arg, Class<T> clazz) {
+    public <T> T run(Target<String> target, String fun, SaltClientType clientType, Class<T> clazz, String... arg) {
         Form form = new Form();
         form = addAuth(form)
                 .param("fun", fun)
                 .param("client", clientType.getType())
                 .param("tgt", target.getTarget())
                 .param("expr_form", target.getType());
         if (arg != null) {
-            form.param("arg", arg);
+            for (String a : arg) {
+                form.param("arg", a);
+            }
+        }
+        T response = saltTarget.path(SaltEndpoint.SALT_RUN
+                .getContextPath()).request()
+                .post(Entity.form(form)).readEntity(clazz);
+        LOGGER.info("Salt run response: {}", response);
+        return response;
+    }
+
+    public <T> T runner(Target<String> target, String fun, String argName, String arg, Class<T> clazz) {
```

Comment:
It seems it is a copy paste of run(..) method. I think it is better to extend the SaltClientType enum with runner